File: lib/models/inventory.dart
/// Model tồn kho
class Inventory {
  final int warehouseId;
  final String warehouseName;
  final int productId;
  final String productName;
  final int quantity;

  Inventory({
    required this.warehouseId,
    required this.warehouseName,
    required this.productId,
    required this.productName,
    required this.quantity,
  });

  factory Inventory.fromJson(Map<String, dynamic> json) => Inventory(
    warehouseId: json['warehouseId'] ?? 0,
    warehouseName: json['warehouseName'] ?? '',
    productId: json['productId'] ?? 0,
    productName: json['productName'] ?? '',
    quantity: json['quantity'] ?? 0,
  );

  Map<String, dynamic> toJson() => {
    'warehouseId': warehouseId,
    'warehouseName': warehouseName,
    'productId': productId,
    'productName': productName,
    'quantity': quantity,
  };
}

File: Models/DataBase/Inventory.cs
namespace quan_ly_chuoi_cua_hang_tap_hoa_api.Models.DataBase
{
    public class Inventory
    {
        public int WarehouseId { get; set; }
        public int ProductId { get; set; }
        public int Quantity { get; set; }

        public virtual Warehouse Warehouse { get; set; }
        public virtual Product Product { get; set; }
    }
}
